build_from.py

```diff
@@ -55,29 +44,45 @@
     builder.gen_and_write_layers(reg_tree, act_title_and_section, layer_cache)
     layer_cache.replace_using(reg_tree)
 
-    
-    # this used to assume implicitly that if gen-diffs was not specified it was
-    # True; changed it to explicit check
     if args.generate_diffs:
-        all_versions = {doc_number: reg_tree}
-
-        for last_notice, old, new_tree, notices in builder.revision_generator(
-                reg_tree):
-            version = last_notice['document_number']
-            logger.info("Version %s", version)
-            all_versions[version] = new_tree
-            builder.doc_number = version
-            builder.write_regulation(new_tree)
-            layer_cache.invalidate_by_notice(last_notice)
-            builder.gen_and_write_layers(new_tree, act_title_and_section,
-                                         layer_cache, notices)
-            layer_cache.replace_using(new_tree)
-
-        # now build diffs - include "empty" diffs comparing a version to itself
-        for lhs_version, lhs_tree in all_versions.iteritems():
-            for rhs_version, rhs_tree in all_versions.iteritems():
-                comparer = treediff.Compare(lhs_tree, rhs_tree)
-                comparer.compare()
-                builder.writer.diff(
-                    reg_tree.label_id(), lhs_version, rhs_version
-                ).write(comparer.changes)
+        generate_diffs(doc_number, reg_tree, act_title_and_section, builder, layer_cache)
+
+def generate_diffs(doc_number, reg_tree, act_title_and_section, builder, layer_cache):
```

Comment:
A docstring here to describe what this function does and why it exists would be useful.